rocketmq-replicator/src/main/java/org/apache/rocketmq/replicator/RmqSinkTask.java

```diff
@@ -0,0 +1,151 @@
+import io.openmessaging.connector.api.data.SinkDataEntry;
+import io.openmessaging.connector.api.sink.SinkTask;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Map;
+import org.apache.rocketmq.acl.common.AclClientRPCHook;
+import org.apache.rocketmq.acl.common.SessionCredentials;
+import org.apache.rocketmq.client.exception.MQClientException;
+import org.apache.rocketmq.client.producer.DefaultMQProducer;
+import org.apache.rocketmq.client.producer.SendResult;
+import org.apache.rocketmq.client.producer.SendStatus;
+import org.apache.rocketmq.common.message.Message;
+import org.apache.rocketmq.common.message.MessageQueue;
+import org.apache.rocketmq.remoting.RPCHook;
+import org.apache.rocketmq.replicator.common.ConstDefine;
+import org.apache.rocketmq.replicator.common.Utils;
+import org.apache.rocketmq.replicator.config.ConfigUtil;
+import org.apache.rocketmq.replicator.config.TaskConfig;
+import org.apache.rocketmq.replicator.schema.FieldName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RmqSinkTask extends SinkTask {
+
+    private Logger log = LoggerFactory.getLogger(RmqSinkTask.class);
+    private final String taskId;
+    private final TaskConfig taskConfig;
+    private DefaultMQProducer producer;
+    private volatile boolean started = false;
+
+    public RmqSinkTask() {
+        this.taskConfig = new TaskConfig();
+        this.taskId = Utils.createTaskId(Thread.currentThread().getName());
+    }
+
+    @Override
+    public void put(Collection<SinkDataEntry> sinkDataEntries) {
+        for (SinkDataEntry dataEntry : sinkDataEntries) {
+            if (this.started) {
+                Schema schema = dataEntry.getSchema();
+                Object[] payload = dataEntry.getPayload();
+
+                String brokerName = (String) getPayloadValue(FieldName.BROKER_NAME.getKey(), schema, payload);
+                String topic = (String) getPayloadValue(FieldName.TOPIC.getKey(), schema, payload);
+                Integer queueId = (Integer) getPayloadValue(FieldName.QUEUE_ID.getKey(), schema, payload);
+                MessageQueue queue = new MessageQueue(topic, brokerName, queueId);
+
+                String offsetMsgId = (String) getPayloadValue(FieldName.OFFSET_MESSAGE_ID.getKey(), schema, payload);
+                String clientMsgId = (String) getPayloadValue(FieldName.CLIENT_MESSAGE_ID.getKey(), schema, payload);
+                String tags = (String) getPayloadValue(FieldName.TAGS.getKey(), schema, payload);
+                String keys = (String) getPayloadValue(FieldName.KEYS.getKey(), schema, payload);
+                String msgBody = (String) getPayloadValue(FieldName.COMMON_MESSAGE.getKey(), schema, payload);
+                Message msg = new Message();
+                msg.setTopic(topic);
+                msg.setBody(msgBody.getBytes(StandardCharsets.UTF_8));
+                msg.setKeys(String.format("%s %s %s", offsetMsgId, clientMsgId, keys));
```

Comment:
The offsetMsgId is added to the keys, which can be used for message offset checking.